### PR TITLE
decorator Cache created

### PR DESCRIPTION
**Pull request by vasyl-main-dev**

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -1,6 +1,16 @@
...
-    pass
+    results = {}
+
+    def inner(*args) -> Any:
+        if args in results:
+            print("Getting from cache")
+            return results[args]
+        print("Calculating new result")
+        temp_res = func(*args)
+        results[args] = temp_res
```
Comment (ArsenPidhoretskyi) [inline app/main.py]: Write this inline. And use if-else statement(return after it)

**Review by ArsenPidhoretskyi — CHANGES_REQUESTED** [review]

File: app/main.py (hunk 1 above)
Comment (vasyl-main-dev) [reply app/main.py]: Sorry, i don't understand. What should i write and how? 
If i write         
 ```
    temp_res = func(*args)
    results[args] = temp_res
```
before if statement i will can't check args in results. 
Also i use return for finishing the function, so i don't need else statement.

Do you mean that i should write 
```
    results[args] = func(*args)
    return results[args]
```
?

Comment (EdAlekseiev) [reply app/main.py]: Yes, I think, that you should write this something like this.
```suggestion
        results[args] = func(*args)
```

**Review by EdAlekseiev — CHANGES_REQUESTED** [review]

**Review by EdAlekseiev — APPROVED** [review]
Good job

**Pull request closed without merging**